src/test/integration/IntegrationBase.t.sol

```diff
@@ -351,6 +347,234 @@ abstract contract IntegrationBase is IntegrationDeployer {
+                        curMagnitude.encumbered, 
+                        prevMagnitude.encumbered, 
+                        string.concat(err, " (encumbered magnitude)")
+                    );
+
+                    assertEq(
+                        curMagnitude.allocatable, 
+                        prevMagnitude.allocatable, 
+                        string.concat(err, " (allocatable magnitude)")
+                    );
+                }
+            }
+
+            // Max magnitude should not have mutated.
+            assertEq(
+                curMagnitude.max, 
+                prevMagnitude.max, 
+                string.concat(err, " (max magnitude)")
+            );
+        }
+    }
+
+    function assert_Snap_Allocations_Slashed(
+        SlashingParams memory slashingParams,
+        OperatorSet memory operatorSet,
+        bool completed,
+        string memory err
+    ) internal {
+        User op = User(payable(slashingParams.operator));
+
+        Allocation[] memory curAllocs = _getAllocations(op, operatorSet, slashingParams.strategies);
+        Allocation[] memory prevAllocs = _getPrevAllocations(op, operatorSet, slashingParams.strategies);
+        Magnitudes[] memory curMagnitudes = _getMagnitudes(op, slashingParams.strategies);
+        Magnitudes[] memory prevMagnitudes = _getPrevMagnitudes(op, slashingParams.strategies);
+
+        (, uint32 delay) = allocationManager.getAllocationDelay(slashingParams.operator);
+
+        for (uint i = 0; i < slashingParams.strategies.length; i++) {
+            Allocation memory curAlloc = curAllocs[i];
+            Allocation memory prevAlloc = prevAllocs[i]; 
+
+            uint64 slashedMagnitude = uint64(uint256(prevAlloc.currentMagnitude).mulWadRoundUp(slashingParams.wadsToSlash[i]));
+
+            assertEq(
+                curAlloc.currentMagnitude, 
+                prevAlloc.currentMagnitude - slashedMagnitude, 
+                string.concat(err, " (currentMagnitude)")
+            );
+
+            if (completed) {
+                assertEq(curAlloc.pendingDiff, 0, string.concat(err, " (pendingDiff)"));
+                assertEq(curAlloc.effectBlock, 0, string.concat(err, " (effectBlock)"));
+            } else {
+                assertEq(
+                    curAlloc.currentMagnitude, 
+                    prevAlloc.currentMagnitude - slashedMagnitude, 
+                    string.concat(err, " (currentMagnitude)")
+                );
+
+                if (prevAlloc.pendingDiff < 0) {
+                    uint64 slashedPending =
+                        uint64(uint256(uint128(-prevAlloc.pendingDiff)).mulWadRoundUp(slashingParams.wadsToSlash[i]));
+                    
+                    assertEq(
+                        curAlloc.pendingDiff, 
+                        prevAlloc.pendingDiff + int128(uint128(slashedPending)), 
+                        string.concat(err, " (pendingDiff)")
+                    );
+                }
+
+                assertEq(
```

Comment:
I think the effectBlock here can vary depending on if we have a pending allocation or deallocation. Need to check for that

Reply:
fixed in [d7a5d61](https://github.com/Layr-Labs/eigenlayer-contracts/pull/934/commits/d7a5d61437737a470dd4d22220e8a818d21c2ac1)